### PR TITLE
Implement `cliponaxis: false` for bar text

### PR DESCRIPTION
**Pull request by etpinard**
Friday Afternoon Feature. This one was talked about in https://github.com/plotly/plotly.js/issues/2000 and https://github.com/plotly/plotly.js/issues/2001 as one way to solve problems with undesired clipped text nodes in bar traces. Note that this PR does not attempt to make on-graph text contribue to the autorange computations.

cc @alexcjohnson @cldougl (and I think this should make @chriddyp happy too :wink: )

### CODE REVIEW

File: src/components/drawing/index.js [hunk 1]
```diff
@@ -100,9 +100,11 @@ drawing.hideOutsideRangePoint = function(d, sel, xa, ya, xcalendar, ycalendar) {
     );
 };
 
-drawing.hideOutsideRangePoints = function(traceGroups, subplot) {
+drawing.hideOutsideRangePoints = function(traceGroups, subplot, selector) {
     if(!subplot._hasClipOnAxisFalse) return;
 
+    selector = selector || '.point,textpoint';
```
Comment (alexcjohnson) [inline src/components/drawing/index.js]: `'.point,.textpoint'` (you lost a `.`)

Comment (alexcjohnson) [reply src/components/drawing/index.js]: do we not have an image test that catches this?

Comment (etpinard) [reply src/components/drawing/index.js]: No. That thing only gets called during pan and zoom. Nice catch. I'll add a pan test to :lock: this down.

Comment (etpinard) [reply src/components/drawing/index.js]: Fixed and :lock: in -> https://github.com/plotly/plotly.js/pull/2378/commits/b9513a22396454123649581c38903e802bceefa3

**Comment by alexcjohnson** [comment]
Very nice, relieves one of the big pain points until we really do text in autorange. 💃

**Comment by chriddyp** [comment]
YESSSSSSSSSS! Love Friday Afternoon Features :1st_place_medal:

**Comment by frankhumnoi** [comment]
Oo

**Pull request merged**